### PR TITLE
Backend/feature/remove user from chat

### LINKED ISSUE #937: Add ability for admins to remove people from group chats
Subject: admins cannot remove people from group chats
Description:
when viewing chat members list as an admin, there is not a way for an admin to remove a member of the chat. This is an essential feature for moderation of group chats. Admins should be able to remove bad actors.

Steps:
Visited both the member list in the group chat and the chat settings.

Results:
Admins should be able to remove members of the group chat.

Backend version: develop-a6cdeee4
Frontend version: develop-a6cdeee4
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:87.0) Gecko/20100101 Firefox/87.0
Page: https://app.couchers.org/messages/chats/71
User ID: 57

### PR DESCRIPTION
**Pull request by Hassan-Mallah**
**Backend checklist**
- [ ] Formatted my code by running `isort . && black .` in `app/backend`
- [ ] Added tests for any new code or added a regression test if fixing a bug
- [x] All tests pass
- [x] Run the backend locally and it works
- [ ] Added migrations if there are any database changes, rebased onto `develop` if necessary for linear migration history

**Frontend checklist**
- [ ] Formatted my code with `yarn format && yarn lint --fix`
- [ ] There are no warnings from `yarn lint`
- [ ] There are no console warnings when running the app
- [ ] Added any new components to storybook
- [ ] Added tests where relevant
- [ ] All tests pass
- [ ] Clicked around my changes running locally and it works
- [ ] Checked Desktop, Mobile and Tablet screen sizes

### CODE REVIEW

**Comment by aapeliv** [comment]
Thanks, this is looking very good!

You should also send a message on that group chat saying an admin removed a user. You'll need to create a new message type in `MessageType` (in `models.py`) and then send it with `_add_message_to_subscription`, and add it to the protobuf enum and `_message_to_pb`. Once implemented in the frontend, it should then render as a message like "Hassan removed Aapeli."

**Comment by Hassan-Mallah** [comment]
> Thanks, this is looking very good!
> 
> You should also send a message on that group chat saying an admin removed a user. You'll need to create a new message type in `MessageType` (in `models.py`) and then send it with `_add_message_to_subscription`, and add it to the protobuf enum and `_message_to_pb`. Once implemented in the frontend, it should then render as a message like "Hassan removed Aapeli."

thank you for the feedback, i'll do as u said. cheers

**Comment by Hassan-Mallah** [comment]
@aapeliv hey,
I did changes as you said, i hope it's all correct :)
there's a test error with migration, i need to change these files as well?
cheers

File: app/backend/src/couchers/models.py [hunk 1]
```diff
@@ -488,8 +488,9 @@ class MessageType(enum.Enum):
     user_invited = enum.auto()
     user_left = enum.auto()
     user_made_admin = enum.auto()
-    user_removed_admin = enum.auto()
+    user_removed_admin = enum.auto()  # RemoveGroupChatAdmin: remove admin permission from a user in group chat
     host_request_status_changed = enum.auto()
+    group_chat_user_removed = enum.auto()  # user is removed from group chat by amdin RemoveGroupChatUser
```
Comment (aapeliv) [inline app/backend/src/couchers/models.py]: let's call this one `user_removed`

File: app/pb/conversations.proto [hunk 2]
```diff
@@ -91,6 +95,11 @@ message MessageContentUserRemovedAdmin {
   uint64 target_user_id = 1;
 }
 
+// field in Message, when a user is removed by admin
+message MessageContentRemoveGroupChatUser {
```
Comment (aapeliv) [inline app/pb/conversations.proto]: `MessageContentUserRemoved`

File: app/backend/src/couchers/servicers/conversations.py [hunk 3]
```diff
@@ -618,6 +621,53 @@ def InviteToGroupChat(self, request, context):
...
+                    .one_or_none()
+            )
+
+            # if user info is missing
+            if not your_subscription:
+                context.abort(grpc.StatusCode.NOT_FOUND, errors.CHAT_NOT_FOUND)
+
+            # if user not admin
+            if your_subscription.role != GroupChatRole.admin:
+                context.abort(grpc.StatusCode.PERMISSION_DENIED, errors.ONLY_ADMIN_CAN_MAKE_ADMIN)
```
Comment (aapeliv) [inline app/backend/src/couchers/servicers/conversations.py]: Please make new error strings in `couchers.errors` and use them here, these don't match the actual errors.

(Also two more instances below)

**Review by aapeliv — CHANGES_REQUESTED** [review]
Great! Just a few minor comments.

I can push a migration file, if you're okay with that? The migration for this has to be done manually because alembic (the migration tool) doesn't detect changes in enums. Also happy for you to do it if you wish and I can point you in the right direction. Either way, you need to rebase onto `develop` (or merge `develop` into this) so we can build the migration on top of the latest schema.

**Comment by Hassan-Mallah** [comment]
@aapeliv Hey,
Thnks for the feedback :)
i did the changes, i think this time you do migration. Maybe next time i'll do it.
plz check. cheers

**Review by aapeliv — APPROVED** [review]
Great work, thanks!

I merged `develop` into here and added the migration. Next time, please remember to run `isort . && black .` in `//app/backend` to format the code.

You can merge as soon as the tests pass.

**Comment by Hassan-Mallah** [comment]
great, thank u ;)

**Pull request merged**